src/main/java/ch/njol/skript/sections/EffSecSendResourcePack.java

```diff
@@ -0,0 +1,214 @@
+	"The URL must be a direct download link.",
+	"",
+	"The [**resource pack request action**](events.html#resource_pack_request_action) can be used to check",
+	"the status of the sent resource pack request.",
+	"",
+	"Take a look at [**Resource Pack Values**](https://docs.skriptlang.org/effects.html#EffResourcePackValues) to see additional options."
+})
+@Examples({
+	"on join:",
+		"\tsend the resource pack from \"URL\" to the player",
+	"",
+		"\tsend the resource pack from \"URL\" to the player using:",
+			"\t\tset the resource pack uuid to \"1\"",
+			"\t\tset the resource pack hash to \"Hash\"",
+			"\t\tset the resource pack prompt to \"Please Download\"",
+			"\t\tforce the player to accept"
+})
+@Since("2.4, INSERT VERSION (section)")
+public class EffSecSendResourcePack extends EffectSection {
+
+	public static class ResourcePackEvent extends Event {
+		@Nullable
+		private String id, hash, prompt;
+		@Nullable
+		private Boolean force;
+
+		public ResourcePackEvent() {}
+
+		public void setId(String id) {
+			this.id = id;
+		}
+		public void setHash(String hash) {
+			this.hash = hash;
+		}
+		public void setPrompt(String prompt) {
+			this.prompt = prompt;
+		}
+		public void setForce(Boolean force) {
+			this.force = force;
+		}
+
+		public String getId() {
+			return id;
+		}
+		public String getHash() {
+			return hash;
+		}
+		public String getPrompt() {
+			return prompt;
+		}
+		public Boolean getForce() {
+			return force;
+		}
+
+
+		@Override
+		@NotNull
+		public HandlerList getHandlers() {
+			throw new IllegalStateException();
+		}
+	}
+
+	static {
+		Skript.registerSection(EffSecSendResourcePack.class,
+			"send [a|the] resource pack [at|from [[the] URL]] %string% to %players% [using|with]",
+			"send [a|the] resource pack [at|from [[the] URL]] %string% with hash %string% to %players%"
+			);
+
+	}
+
+	@UnknownNullability
+	@SuppressWarnings("null")
+	private Expression<String> url;
+	@UnknownNullability
+	@SuppressWarnings("null")
+	private Expression<Player> recipients;
+	@Nullable
+	private Trigger trigger;
+	@Nullable
+	private Expression<String> oldHash;
+	private int pattern;
+
+	@SuppressWarnings({"unchecked", "null"})
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult, @Nullable SectionNode sectionNode, List<TriggerItem> triggerItems) {
+		this.url = (Expression<String>) exprs[0];
+		this.pattern = matchedPattern;
+		if (matchedPattern == 2) {
+			this.oldHash = (Expression<String>) exprs[1];
+			this.recipients = (Expression<Player>) exprs[2];
+		} else {
+			this.recipients = (Expression<Player>) exprs[1];
+		}
+
+		if (sectionNode != null) {
+			AtomicBoolean delayed = new AtomicBoolean(false);
+			Runnable afterLoading = () -> delayed.set(!getParser().getHasDelayBefore().isFalse());
+			trigger = loadCode(sectionNode, "resource", afterLoading, ResourcePackEvent.class);
```

Comment:
this is a bit dangerous honestly. I don't think we should be allowing just any syntax within the section. Otherwise you get:
```
send resource pack...:
    spawn villager at location():
        send resource pack...:
            message "test" to all players
```

Reply:
So check to see if the parser current section is null? if not, error?

Reply:
no i think this needs a different approach
I'll get back to you after some investigation

Reply:
@sovdeeth This fix that you're constructing, is it gonna be added in this PR or another/new PR specifically for it?

Reply:
So there's two options here:
a) leave it as is and let people get into trouble if they want to
b) only attempt to parse the stuff in the section as EffResourcePackValues, so they can't get into trouble even if they wanted to.

I think b's probably the better choice here but it's a bit more complicated than what you have right now.